### PR TITLE
VUU71: Connect remote implementation of application layouts to server

### PR DESCRIPTION
**Pull request by pling-scottlogic**
### Description
When the app is configured to use remote layout persistence, saving and loading application layouts currently has no effect. The changes here allow it to communicate with the layout server, so the application layouts are saved to and retrieved from the database.

### Change List
- Implement fetches from server for saving and loading application layouts in `RemoteLayoutPersistenceManager`
- Add `ApplicationLayout` type to model the server's response body
- Implement separate catches for metadata and application layout retrieval in `useLayoutManager` (plus refactor)
- Remove logging from local layout manager

### CODE REVIEW

File: vuu-ui/packages/vuu-layout/src/layout-persistence/RemoteLayoutPersistenceManager.ts [hunk 1]
```diff
@@ -134,13 +138,49 @@ export class RemoteLayoutPersistenceManager
...
+      })
+      .then((response) => {
+        if (!response.ok) {
+          reject(new Error(response.statusText));
+        }
+        response.json().then((response: ApplicationLayout) => {
+          if (!response) {
+            reject(new Error("Response did not contain valid application layout information"));
+          }
+          resolve(response.definition);
```
Comment (pling-scottlogic) [inline vuu-ui/packages/vuu-layout/src/layout-persistence/RemoteLayoutPersistenceManager.ts:178]: Since we're only ever using the definition, it's probably worth changing the data contract to remove the username from the response. I've raised a [ticket](#102) for this to avoid conflicts with ongoing work (and scope creep).

File: vuu-ui/packages/vuu-layout/src/layout-persistence/RemoteLayoutPersistenceManager.ts [hunk 2]
```diff
@@ -134,13 +138,49 @@ export class RemoteLayoutPersistenceManager
...
+        method: "GET",
+        headers: {
+          "username": "vuu-user",
+        },
+      })
+      .then((response) => {
+        if (!response.ok) {
+          reject(new Error(response.statusText));
+        }
+        response.json().then((response: ApplicationLayout) => {
```
Comment (vferraro-scottlogic) [inline vuu-ui/packages/vuu-layout/src/layout-persistence/RemoteLayoutPersistenceManager.ts:174]: I would name `response` something different here to differentiate from the `response` on line 170 and to make it clearer that thats the content of the response body rather than the whole response. Maybe `applicationLayout`?

**Review by vferraro-scottlogic — COMMENTED** [review]
Looks good! Just 1 suggestion around naming

File: vuu-ui/packages/vuu-layout/src/layout-persistence/RemoteLayoutPersistenceManager.ts (hunk 2 above)
Comment (pling-scottlogic) [reply vuu-ui/packages/vuu-layout/src/layout-persistence/RemoteLayoutPersistenceManager.ts]: Good point, I didn't notice they were different. Renamed here: 14f493b.

**Review by vferraro-scottlogic — APPROVED** [review]
